src/ucs/memory/memtype_cache.c

```diff
@@ -117,11 +117,11 @@ UCS_PROFILE_FUNC_VOID(ucs_memtype_cache_update_internal,
         /* try to find regions that are contiguous and instersected
          * with current one */
         search_start = start - 1;
-        search_end   = end + 1;
+        search_end   = end;
```

Comment:
good catch (should have seen it in code review..)
can u pls update unit test to catch this?

Reply:
@yosefe can we simulate this situation through generating UCM events?

Reply:
we should be able to simulate this with memtype_cache API functions only
i think you already have some tests doing that..